packages/css/search.css

```diff
@@ -1,170 +1,108 @@
-  padding: 0 var(--ds-spacing-4);
-  padding-right: 2.2em;
-}
-
-.ds-search--md .ds-search__icon {
-  left: var(--ds-spacing-4);
-}
+  /* The simple variant, when no submit button is present */
+  &:not(:has([type='submit'])) {
+    & input {
+      padding-left: 2.4em;
+    }
+
+    &::before {
+      content: '';
+      position: absolute;
+      left: var(--ds-sizing-2);
+      top: 50%;
+      transform: translateY(-50%);
+      height: var(--dsc-search-icon-size);
+      aspect-ratio: 1 / 1;
+      mask: var(--dsc-search-input-icon) center / contain no-repeat;
+      background: currentcolor;
+      z-index: 2;
+      pointer-events: none;
+    }
+  }
+  & input {
+    grid-column: 1 / 2;
+    grid-row: 1 / 2;
+
+    &:focus-visible {
+      z-index: 1;
+    }
+    &::-webkit-search-decoration,
+    &::-webkit-search-cancel-button {
+      appearance: none;
+    }
+  }
 
-.ds-search--lg .ds-search__input {
-  --dsc-search-clear-button-size: var(--ds-sizing-12);
+  &:has(input:placeholder-shown) [type='reset'] {
+    display: none;
+  }
 
-  height: var(--ds-sizing-14);
-  padding: 0 var(--ds-spacing-5);
-  padding-right: 2em;
-}
+  & [type='reset'] {
+    --dsc-button-padding: 0;
+
+    grid-column: 1 / 2;
+    grid-row: 1 / 2;
+    justify-self: end;
+    align-self: center;
+    margin-right: var(--ds-spacing-3);
+    border: none;
+    cursor: pointer;
+    height: var(--dsc-search-clear-button-size);
+    width: var(--dsc-search-clear-button-size);
+    border-radius: var(--ds-border-radius-default);
+    font-size: 0.5em;
+    z-index: 2;
+    position: relative;
+
+    &::before {
+      content: '';
+      position: absolute;
+      left: 50%;
+      top: 50%;
+      transform: translate(-50%, -50%);
+      height: var(--dsc-search-clear-icon-size);
+      aspect-ratio: 1 / 1;
+      mask: var(--dsc-search-clear-icon) center / contain no-repeat;
+      background: currentcolor;
+      z-index: 2;
+      pointer-events: none;
+    }
+  }
 
-.ds-search--lg .ds-search__icon {
-  left: var(--ds-spacing-5);
-}
+  & [type='submit'] {
+    border-top-left-radius: 0;
+    border-bottom-left-radius: 0;
 
-.ds-search__input.ds-search__input--simple {
-  padding-left: 2.4em;
-}
+    &:focus-visible {
+      z-index: 1;
+    }
 
-@media (hover: hover) and (pointer: fine) {
-  .ds-search__input:not(:focus-visible, :disabled, [aria-disabled]):hover {
-    --dsc-search-input-border-color: var(--ds-color-accent-border-strong);
+    &:not(:focus-visible) {
+      border-left: 0;
+    }
+  }
 
-    box-shadow: inset 0 0 0 1px var(--ds-color-accent-border-strong);
+  &:has([type='submit']) {
+    & input {
+      border-top-right-radius: 0;
+      border-bottom-right-radius: 0;
+    }
   }
 
-  .ds-search__clear-button:not(:focus-visible, :disabled, [aria-disabled]):hover {
-    background: var(--ds-color-accent-surface-hover);
+  @media (hover: hover) and (pointer: fine) {
+    [type='reset']:not(:focus-visible, :disabled, [aria-disabled]):hover {
```

Comment:
Here, the :hover is not added if `button[type="reset"]` is `:disabled` - was that the intention, or was it when `input:disabled`? ☺️

Reply:
This is just what was there before.

Reply:
We should investigate then ☺️ 
seems a bit redundant to me? and wrong as `[aria-disabled="false"]` also exists ☺️ 
Maybe we instead should have a:
```
.ds-search:has(:disabled, [aria-disabled="true"]) button[type="reset"] { display: none }
```
so we can not clear if disabled? ☺️